lib/attr_vault/secret.rb

```diff
@@ -31,11 +37,19 @@ def initialize(secret)
 
     # Internal: Returns the portion of the secret token used for encryption
     def encryption_key
-      @secret.slice(16, 16)
+      if @aead
+        @secret
+      else
+        @secret.slice(16, 16)
+      end
     end
 
     # Internal: Returns the portion of the secret token used for signing
     def signing_key
+      if @aead
+        raise Error.new('signing_key must not be used for AEAD ciphers')
```

Comment:
Okay, so stupid question: is this because... the HMAC behavior is built into the new mechanism?

Reply:
Yeah. Through maf, you don't need any more bytes other than the AES key itself, somehow involving the IV, which it is *very important* not to reuse. auth_tag is what it's called, and another term, when referring to the GCM implementation in particular, is "GMAC".